### DISCUSSION TITLE
Get all productTypes

### DISCUSSION THREAD

**Discussion by marc-costea** [amzn/selling-partner-api-models#3837] [General]
Hi all,

I am pretty new to the amazon sp-api so please forgive me if the question is stupid ;-)

Backgroud:
We want to show our users a category tree and let them decide in which category they want to list (because this is common sense in our UI with other marketplaces) and after choosing the category, we are importing the necessary attributes into their local systems (and let them fill out the necessary information).

Therefore I need to get a full category list with its "productType" ID's, but when I'm try to get it from 
 [`/definitions/2020-09-01/productTypes`](https://developer-docs.amazon.com/sp-api/docs/product-type-definitions-api-v2020-09-01-reference#get-definitions2020-09-01producttypes)
 (without any keywords, the only parameter is the marketplaceId for Germany) I receive only 1 result:
```json
{
  "productTypes": [
    {
      "name": "LUGGAGE",
      "displayName": "Luggage",
      "marketplaceIds": [
        "ATVPDKIKX0DER"
      ]
    }
  ],
  "productTypeVersion": "UHqSqmb4FNUk="
}
```

How can I get every productType (with at least its ID) by the sp-api?

**Comment by marc-costea** [comment]
found the solution (if anybody has the same problem):

If your are requesting definitions/{version}/productTypes API - do NOT add suffix "sandbox." in your subdomain.
with for example `https://sandbox.sellingpartnerapi-na.amazon.com/definitions/2020-09-01/productTypes/` you will only receive the luggage category. and if you call for details you will receive an invalid schema url ("https://schema-url"). Without the "sandbox." suffix (also if your app is in draft state!) you will receive every productType and also valid details with valid schema urls.

**Comment by mafge** [comment]
Hi @marc-costea,
thanks for reaching out!
I am happy that you were able to find a solution for your problem. Please note that the sandbox in most cases returns limited static data, if you need production data it is not recommended to use the sandbox environment. There is also another way to get all product types which also gives information about the hierarchy of each product type which is the [Browse Tree Report](https://developer-docs.amazon.com/sp-api/docs/report-type-values-browse-tree). Please also note that product types can differ from marketplace to marketplace, so make sure to check for each marketplace individually.
Best,
Marc
Selling Partner Developer Services

**Comment by marc-costea** [comment]
Hi @mafge ,

thats also way we have implemented it.

First getting the Browse Nodes Tree (also to show our users the hierarchy of the nodes) and getting each possible productType for a specific node.
Then getting the details with the `/definitions/{AmazonRequest.VERSION}/productTypes/{{productType}}` request and downloading the scheme file (for create and update products and listings) and every possible attribut + its information (description, if its required or not ....) to display it to the user.

Hopefully our discussion will help and save time other devs in the future reading this =)
